src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,793 @@
+  //                          Resize implementation
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+  #handleEdgeResize = (edge: ChDialogResizeElement) => (event: MouseEvent) => {
+    this.#fixDialogPosition();
+
+    // this.resizing = true;
+    this.#resizeRAF ||= new SyncWithRAF();
+    this.#currentEdge = edge;
+    this.#initialDragEvent = event;
+
+    // Specify the cursor for the resize operation. Useful to avoid showing
+    // incorrect cursors during resizing
+    resizingCursorDictionary[this.#currentEdge](this.#isRTLDirection);
+
+    // Initialize drag variables to improve block-start and inline-start
+    // resizing. Otherwise, the dialog will always remain in the same X and Y
+    // position, even when the block-start or inline-start edges are resized
+    this.#draggedDistanceXForResize = this.#draggedDistanceX;
+    this.#draggedDistanceYForResize = this.#draggedDistanceY;
+
+    // Get minimum and maximum sizes on first resize operation
+    const computedStyle = getComputedStyle(this.#dialogRef);
+    this.#maxBlockSize = fromPxToNumber(computedStyle.maxBlockSize);
+    this.#maxInlineSize = fromPxToNumber(computedStyle.maxInlineSize);
+    this.#minBlockSize = fromPxToNumber(computedStyle.minBlockSize);
+    this.#minInlineSize = fromPxToNumber(computedStyle.minInlineSize);
+
+    // Avoid watching border changes during the resize
+    this.#removeBorderSizeWatcher();
+
+    // Avoid listener on document click
+    document.removeEventListener("click", this.#evaluateClickOnDocument, {
+      capture: true
+    });
+
+    // Add listeners
+    document.addEventListener("mousemove", this.#trackElementResizeRAF, {
+      capture: true
+    });
+
+    document.addEventListener("mouseup", this.#handleResizeEnd, {
+      capture: true,
+      passive: true
+    });
+  };
+
+  #trackElementResizeRAF = (event: MouseEvent) => {
+    this.#resizeRAF.perform(this.#trackElementResize, () => {
+      // Improve drag UX by not selecting any button or clicking interactive
+      // elements
+      event.preventDefault();
+
+      // We remove the pointer-events and user-select properties after the first
+      // "mousemove", otherwise double clicking to select text would not work
+      this.#addDraggingClass();
+
+      this.#lastDragEvent = event;
+    });
+  };
+
+  #trackElementResize = () => {
+    // - - - - - - - - - - - - - DOM read operations - - - - - - - - - - - - -
+    const dialogRect = this.#dialogRef.getBoundingClientRect();
+
+    // - - - - - - - - - - - - - DOM write operations - - - - - - - - - - - - -
+    this.#resizeEdgesAndCornersDictionary[this.#currentEdge](dialogRect);
+
+    // Update last point
+    this.#initialDragEvent = this.#lastDragEvent;
+  };
+
+  #handleResizeEnd = () => {
+    if (this.adjustPositionAfterResize) {
+      this.#unfixDialogPosition();
+    } else {
+      this.#fixDialogPosition();
+    }
```

Comment:
Why do we need the "else" case?

Reply:
It only seems to be required to fix the dialog position if _this.adjustPositionAfterResize_ is false.

>